tests/utils/test_encrypted_file_metadata.py

```diff
@@ -0,0 +1,92 @@
+#  Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import copy
+import unittest
+
+from matrix_content_scanner.utils.constants import ErrCode
+from matrix_content_scanner.utils.encrypted_file_metadata import (
+    validate_encrypted_file_metadata,
+)
+from matrix_content_scanner.utils.errors import ContentScannerRestError
+from tests.testutils import ENCRYPTED_FILE_METADATA
+
+
+class EncryptedMetadataValidationTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.metadata = copy.deepcopy(ENCRYPTED_FILE_METADATA)
+
+    def test_validate(self) -> None:
+        """Tests that valid file metadata is considered as such."""
+        validate_encrypted_file_metadata(ENCRYPTED_FILE_METADATA)
+
+    def test_key_ops_no_decrypt(self) -> None:
+        """Tests that the metadata validation fails if key_ops doesn't include `decrypt`."""
+        self.metadata["file"]["key"]["key_ops"] = ["encrypt"]
+        self._test_fails_validation()
+
+    def test_key_ops_no_encrypt(self) -> None:
```

Comment:
Maybe a test if there's an extra op would also make sense?

Reply:
Having an extra op would be technically correct. The spec says:

> Must at least contain `encrypt` and `decrypt`.

And all we want to be sure about is that we can use the key for decrypting the file.

Reply:
> Having an extra op would be technically correct. The spec [says](https://spec.matrix.org/v1.2/client-server-api/#extensions-to-mroommessage-msgtypes):

I'm wondering if we should be sure that the code handles that case and doesn't reject it?

Reply:
Ah gotcha. Yes that sounds like a good idea.

Reply:
Done in https://github.com/matrix-org/matrix-content-scanner-python/pull/25/commits/1876b0c54f65ac3f70f57dcdb99dc83b06f1f662